packages/playground/website/public/plugin-proxy.php

```diff
@@ -338,17 +338,56 @@ function ($curl, $body) use (&$extra_headers_sent, $default_response_headers) {
         $branch = strtolower( $_GET['wordpress-branch'] );
         if ( $branch === 'trunk' || $branch === 'master' ) {
             $branch = 'master';
+            $prefix = 'refs/heads/';
             // If the brach is of the form x.x append '-branch' to it.
         } elseif ( preg_match( '/^\d+\.\d+$/', $branch ) ) {
             $branch .= '-branch';
-        } elseif ( ! preg_match( '/^\d+\.\d+-branch$/', $branch ) ) {
-            throw new ApiException( 'Invalid branch' );
+            $prefix = 'refs/heads/';
+            // If the branch is in the form x.x.x, it's a tag.
+        } elseif ( preg_match( '/^\d+\.\d+\.\d+$/', $branch ) ) {
+            // Remove trailing .0 if present.
+            if ( substr( $branch, -2 ) === '.0' ) {
+                $branch = substr( $branch, 0, -2 );
+            }
+            $prefix = 'refs/tags/';
+            // If the branch is in the form [a-f0-9]{7,40} it's a commit hash.
+        } elseif ( preg_match( '/^[a-f0-9]{7,40}$/', $branch ) ) {
```

Comment:
is there a compelling reason to arbitrarily constrain this to seven characters?
`git` itself imposes a minimum length of 4 characters for short ids. [core.abbrev](https://git-scm.com/docs/git-config#Documentation/git-config.txt-coreabbrev)

https://git-scm.com/book/en/v2/Git-Internals-Git-References

Reply:
@dmsnell GitHub displays short hashes as seven characters. I can change it to six if you wish but anything shorter will throw a 404.

Five characters: https://github.com/WordPress/WordPress/commit/eac35
Six characters: https://github.com/WordPress/WordPress/commit/eac356

Reply:
@peterwilsoncc Hm. well it looks like Github is the one pushing arbitrary limits. Still, it's not a guarantee that seven characters will return a real SHA if there's a seven-character clash, as is the case for these:

 - https://github.com/WordPress/WordPress/commit/733e06315a (works)
 - https://github.com/WordPress/WordPress/commit/733e06360c (works)
 - https://github.com/WordPress/WordPress/commit/733e063 (404)

We can't resolve that though since the problem is their system. `git` will display as many characters as it needs to resolve clashes, with a minimum of four characters. Github apparently displays seven even if that's ambiguous.

Reply:
> Github apparently displays seven even if that's ambiguous.

Hilarious

In my WordPress/WordPress checkout `git rev-parse --short HEAD` shows ten characters, `b428e1b5be`, so I guess that's the minimum required to avoid conflicts.

What's your recommendation here, leave as is, increase to 10 or require the full hash?

Reply:
@peterwilsoncc I think the main point I was trying to make was that short refs are just a display thing and not inherent. what works today might not work tomorrow. if you set `core.abbrev = 4` and re-run `git rev-parse --short HEAD` it should show four characters.

this is all fine in the PR as suggested: I didn't realize Github imposed the restrictions it does.